syntax = "proto3";

import "src/semantic_model/system_state/proto/static_scene.proto";
import "src/semantic_model/system_state/proto/dynamic_state.proto";

package semantic_model;


message VehicleDynamicStateSequence {
  repeated VehicleDynamicState states = 1;
}

message TrafficLightDynamicStateSequence {
  repeated TrafficLightDynamicState states = 1;
}

message DynamicSequences {
  map<string, VehicleDynamicStateSequence> vehicles_dynamic_state_sequences = 1;
  map<string, TrafficLightDynamicStateSequence> traffic_lights_dynamic_state_sequences = 2;
}

message DynamicSettings {
  map<string, VehicleItineraryBase> vehicle_itinerary_bases = 1;
}

message SystemRunHeader {
  uint64 length = 1;
  double delta_t = 2;
  repeated string vehicle_ids = 3;
}

message SystemRun {
  SystemRunHeader header = 1;
  StaticScene static_scene = 2;
  DynamicSettings dynamic_settings = 3;
  DynamicSequences dynamic_sequences = 4;
}
